earthmover/earthmover.py

```diff
@@ -139,7 +139,13 @@ def load_project_configs(self, filepath: str):
             if isinstance(val, str):
                 # Combine `key: ${VAL}` with `VAL: default_val` to get `key: default_val`
                 template = string.Template(val)
-                configs[key] = template.substitute(self.params)
+                try:
+                   # Especially in cases like 'deps' and 'compile,' there may be no params passed 
+                   # and no default defined. We should allow this but alert the user so that
+                   # any downstream errors during 'run' are more traceable
+                   configs[key] = template.substitute(self.params)
+                except KeyError:
+                    self.logger.warning(f"No value passed for config.{key}: {val} and no default defined")
```

Comment:
I believe the behavior in the rest of `earthmover.yml` is to ignore params that are not defined, so
```yaml
sources:
  my_file:
    file: ./${NOT_DEFIEND}/file.csv
```
would compile to
```yaml
sources:
  my_file:
    file: ./${NOT_DEFIEND}/file.csv
```
if `NOT_DEFINED` is, well, not defined.

We could achieve the same behavior with `template.safe_substitute(self.params)` here in the `config` section.

It may be that we want to warn about params that are not defined, but I think the behavior should be the same whether they occur in the `config` section or elsewhere.

Reply:
Oh wow okay, I didn't know about `safe_substitute` - that seems like clearly the better choice here. Just updated